relay-metrics/src/aggregation.rs

```diff
@@ -745,6 +754,31 @@ impl BucketKey {
         std::hash::Hash::hash(self, &mut hasher);
         hasher.finalize() as i64
     }
+
+    /// Remove invalid characters from tags and metric names.
+    ///
+    /// Returns `Err` if the metric should be dropped.
+    fn validate_strings(mut self) -> Result<Self, ()> {
+        if self.metric_name.contains('\0') {
+            relay_log::error!("invalid metric name {:?}", self.metric_name);
+            return Err(());
```

Comment:
Let's not log here but rather return `AggregateMetricsError` that can be logged somewhere else.

Reply:
I didn't go for that because it would bloat the size of `AggregateMetricsError` but I can do it

Reply:
Ah, I see. Maybe log this on `debug!` then, we certainly don't want that in a production instance.

Reply:
I have removed it entirely now :/